.github/workflows/ci.yml

```diff
@@ -90,6 +90,50 @@ jobs:
           path: reports-unit-tests.zip
           if-no-files-found: ignore
 
+  crypto-tests:
+    name: Crypto Tests
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: ${{ env.DEFAULT_JDK_VERSION }}
+          cache: 'maven'
+      - name: Update maven settings
+        run: mkdir -p ~/.m2 ; cp .github/settings.xml ~/.m2/
+      - name: Cleanup org.keycloak artifacts
+        run: rm -rf ~/.m2/repository/org/keycloak >/dev/null || true
+      - name: Download built keycloak
+        id: download-keycloak
+        uses: actions/download-artifact@v3
+        with:
+          path: ~/.m2/repository/org/keycloak/
+          name: keycloak-artifacts.zip
+      - name: Run crypto tests (BCFIPS non-approved mode)
```

Comment:
@mposolda Great! Just nitpick regards optimization.
Is it possible to use the matrix for the approved/non-approved mode in order to execute those tests in parallel? For instance, the base testsuite uses that.

Reply:
@mabartos Thanks for the feedback. I am aware of the matrix, but in this case it looked to me like unnecessary overhead as the unit tests itself are very fast to run (usually 10-15 seconds) and hence the matrix will be rather overhead due the pre-steps (like "Download built Keycloak"), which themselves need to be run for each part of the matrix independently.

Maybe we can consider add matrix later if the unit tests grow to the level when they take more significant time to run, however not sure if it happens for the "crypto" modules. WDYT?

Reply:
@mposolda You're right, agree. Thanks!